src/main/scala/com/microsoft/kusto/spark/utils/DeviceAuthentication.scala

```diff
@@ -0,0 +1,67 @@
+package com.microsoft.kusto.spark.utils
+
+import java.awt.Desktop
+import java.net.URI
+import java.util.concurrent.{ExecutionException, ExecutorService, Executors, Future}
+
+import com.microsoft.aad.adal4j.{AuthenticationContext, AuthenticationResult, DeviceCode}
+import javax.naming.ServiceUnavailableException
+import org.apache.log4j.{Level, Logger}
+import java.awt.Toolkit
+import java.awt.datatransfer.StringSelection
+import java.awt.datatransfer.DataFlavor
+
+
+object DeviceAuthentication {
+  private val CLIENT_ID = "db662dc1-0cfe-4e1c-a843-19a68e65be58"
+
+  def acquireAccessTokenUsingDeviceCodeFlow(clusterUrl: String, authority: String = "common") : String =
+  {
+    val aadAuthorityUri = s"https://login.microsoftonline.com/$authority"
+    val service: ExecutorService =
+    Executors.newSingleThreadExecutor
+    val context: AuthenticationContext =  new AuthenticationContext(aadAuthorityUri, true, service)
+
+    var result: AuthenticationResult = null
+    val future: Future[DeviceCode] = context.acquireDeviceCode(CLIENT_ID, clusterUrl, null)
+    val deviceCode: DeviceCode = future.get
+    val clipboard = Toolkit.getDefaultToolkit.getSystemClipboard
+    val dataFlavor = DataFlavor.stringFlavor
+    var text: String = null
+    if (clipboard.isDataFlavorAvailable(dataFlavor)) {
+      text = clipboard.getData(dataFlavor).asInstanceOf[String]
+    }
+
+    clipboard.setContents(new StringSelection(deviceCode.getUserCode), null)
+    println(deviceCode.getMessage)
```

Comment:
Remove (I guess - or KDSU.logInfo in which case add some description of what is logged)

Reply:
it is already with description
and its not a log - i print to the user what he should do next - if he plays with logs verbosity he won't see a log info and anyway it is not